dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -67,17 +68,81 @@ Block HashJoinProbeBlockInputStream::getHeader() const
 
 Block HashJoinProbeBlockInputStream::readImpl()
 {
-    if (probe_process_info.all_rows_joined_finish)
+    // if join finished, return {}
+    if (join_finished)
+    {
+        return Block{};
+    }
+
+    Blocks result_blocks;
+    size_t output_rows = 0;
+
+    // if over_limit_block is not null, we need to push it into result_blocks first.
+    if (over_limit_block)
+    {
+        result_blocks.push_back(over_limit_block);
+        output_rows += over_limit_block.rows();
+        over_limit_block = Block{};
```

Comment:
can we try to use push_back(std::move(over_limit_block)) here?

Reply:
done.

Reply:
After std::move, no need to reset over_limit_block here.

Reply:
> After std::move, no need to reset over_limit_block here.

I think it may be more safe, you can reference https://stackoverflow.com/questions/9168823/reusing-a-moved-container